### DISCUSSION TITLE
Question about Lesson: 25. Debugging the fuzz tests handler(DEFi section)

### DISCUSSION THREAD

**Discussion by nikita-biichuk** [Cyfrin/foundry-full-course-cu#3108] [General]
In this lesson, we debugged the mintDsc handler by adding a check to make sure that the address has already deposited collateral before minting. However, I don’t understand why we didn’t apply the same check to the redeemCollateral handler. Patrick also didn’t mention this point.

Isn’t it true that we can only redeem collateral if we’ve deposited it first?

**Comment by EngrPips** [comment]
That is true and we should add the same check .

**Reply by nikita-biichuk** [reply]
Thanks for confirming!

**Reply by EngrPips** [reply]
Anytime buddy.
